docs/add-your-app.md

```diff
@@ -0,0 +1,138 @@
+|imageVersion|It is recommended to use a specific docker tag for the image, otherwise use "latest"|
+|environmentVariables|Environment variables expected by the container. it is possible to inject dependency bindings into the variable. In our case we injected the mongoDB server address as part of the environment variable value  
+|dependencies|Array of infrastructure service dependencies of the app service|
+|exposedPorts| Array of ports the app service expects kubernetes to expose externally|
+|httpPort| In case of multiple ports exposed, which port should be used for http access (if applicable). Kubernetes will expose this port externally as port 80 if possible|
+|entryPointUrl|Relative path for entry point url for the service|
+
+<br/>
+
+For each dependency of the app service, add an entry under "dependencies". the dependencies format:
+
+<br/>
+
+|property|description|
+|---|---|
+|type|type of the dependency, used for display only DATABASE,OBJECTSTORE,MESSAGING,CACHE,SCHEDULER,LOGGING,VOLUME,OTHER
+|name|dependency display name|
+|description|Additional description for the dependency and what it is used for|
+|protocols|array supported protocols. This is an array since some apps support alternative protocols (e.g. can use either postgres or mysql according to what is available)|
+
+<br/>
+
+The dependency protocol format:
+
+<br/>
+
+|property|description|
+|---|---|
+|protocolName|name of the protocol - will be matched with DSB supported protocols. e.g. mongodsb,postgres,mysql,s3,docker-volume...|
+|version| optional - use the version field if the service required a minimal version of the protocol
+|conditions| optional - the conditions can be used to restrict protocol attributes for specific protocol requirements 
+
+<br/>
+
+## Generic App Templates
+
+Application templates are built to as generic as possible so the apps could run on as many different clusters as 
+possible. For example:
+- External dependencies are described as dependencies of protocols and not specific services so that dependency could 
+be satisfied by different vendor services. If the service requires S3 in order to store documents, we'll depend on 
+the S3 protocol and not the AWS S3 service - this way the application can run on other environments that support the 
+popular S3 protocol (e.g. minio)
+- Templates include only the docker image names and not the docker registry they should be taken from. each cluster 
+can be configured with different docker registries private or public
+
+## Adding App Template Icon
+
+In order to make Ocopea display the application template icon in the Ocopea UI, you'll need to upload the application 
+template icon via the ocopea API. To do that you'll need the appTemplateId returned by the create-app-template command.
+You can find the appTemplateId doing an GET command on the ocopea api on url hub-web-api/app-template.
```

Comment:
"You can find the appTemplateId doing an GET command on the ocopea api on url hub-web-api/app-template." ->
"You can find the appTemplateId by executing a GET command on the ocopea api on url hub-web-api/app-template."